cli/bptest/metadata_lint.go

```diff
@@ -0,0 +1,114 @@
+package bptest
+
+import (
+	"fmt"
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/cli/bpmetadata"
+	"google.golang.org/protobuf/encoding/protojson"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sigs.k8s.io/yaml"
+)
+
+// RunMetadataLintCommand is the entry function that will run the metadata.yml lint checks.
+func RunMetadataLintCommand() {
+	metadataFile := findMetadataFile()
+	if metadataFile == "" {
+		fmt.Println("Error: metadata.yaml file not found")
+		os.Exit(1)
+	}
+
+	metadata, err := ParseMetadataFile(metadataFile)
+	if err != nil {
+		fmt.Printf("Error parsing metadata file: %v\n", err)
+		os.Exit(1)
+	}
+
+	ctx := LintContext{
+		Metadata: metadata,
+		FilePath: metadataFile,
+	}
+
+	runner := &LintRunner{}
+	runner.RegisterRule(&BlueprintVersionRule{})
+
+	// Run lint checks
+	errs := runner.Run(ctx)
+	if len(errs) > 0 {
+		fmt.Println("Linting failed with the following errors:")
+		for _, err := range errs {
+			fmt.Println("- ", err)
+		}
+		os.Exit(1)
+	} else {
+		fmt.Println("All lint checks passed!")
+	}
+}
+
+// findMetadataFile searches for 'metadata.yaml' in the current directory or parent directories.
+func findMetadataFile() string {
+	dir, err := os.Getwd()
+	if err != nil {
+		fmt.Printf("Error getting current directory: %v\n", err)
+		os.Exit(1)
+	}
+
+	for {
+		metadataFilePath := dir + "/metadata.yaml"
+		if _, err := os.Stat(metadataFilePath); err == nil {
+			return metadataFilePath
+		}
+
+		parentDir := filepath.Dir(dir)
+		if parentDir == dir {
+			break
+		}
+		dir = parentDir
+	}
+
+	return ""
+}
+
+// ParseMetadataFile reads a YAML file, converts it to JSON, and unmarshals it into a proto message.
+func ParseMetadataFile(filePath string) (*bpmetadata.BlueprintMetadata, error) {
```

Comment:
Can we reuse this 
https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/blob/d4a376257619e799b26f9324572b506b74f29fc6/cli/bpmetadata/cmd.go#L454